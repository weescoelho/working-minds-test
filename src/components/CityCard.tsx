import { Button } from '@material-ui/core';
import { api } from 'api/api';
import { useEffect, useState } from 'react';
import styled from 'styled-components';
import { State } from 'utils/types';

const CardContainer = styled.div`
  margin: 8px 0;
  background-color: #fff;
  border-radius: 5px;
  padding: 1rem;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  a {
    text-decoration: none;
    color: ${({ theme }) => theme.colors.text};
  }
  button {
    margin-left: 8px;
  }
`;

type StateCardProps = {
  cityId: string;
  cityName: string;
};

export function CityCard({
  cityId,
  cityName = 'SÃ£o Paulo',
}: StateCardProps): JSX.Element {
  const handleDeleteCity = (): void => {
    api.delete(`/cities/${cityId}`);
  };

  return (
    <CardContainer>
      <p>{cityName}</p>
      <div>
        <Button variant="contained" color="primary">
          Editar
        </Button>
        <Button
          variant="contained"
          color="secondary"
          onClick={handleDeleteCity}
        >
          Excluir
        </Button>
      </div>
    </CardContainer>
  );
}
