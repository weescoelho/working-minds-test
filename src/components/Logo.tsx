import { Link } from 'react-router-dom';
import styled from 'styled-components';

const LogoContainer = styled.div`
  font-size: 1.6rem;
  font-weight: 700;
  color: '#F4F4F4';
  text-decoration: none;
`;

export function Logo(): JSX.Element {
  return <LogoContainer>CRUD</LogoContainer>;
}
