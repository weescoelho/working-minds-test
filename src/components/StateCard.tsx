import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { City } from 'utils/types';

const CardContainer = styled.div`
  background-color: #fff;
  max-width: 300px;
  border-radius: 5px;
  padding: 1rem;
  cursor: pointer;
  a {
    text-decoration: none;
    color: ${({ theme }) => theme.colors.text};
  }
`;

type StateCardProps = {
  id: string;
  stateName: string;
  cities: Array<City>;
};

export function StateCard({
  id,
  stateName = 'SÃ£o Paulo',
  cities,
}: StateCardProps): JSX.Element {
  return (
    <CardContainer>
      <Link to={`/estado/${id}`}>
        <h3>{stateName}</h3>
        <p>Cidades cadastradas: {cities.length}</p>
      </Link>
    </CardContainer>
  );
}
