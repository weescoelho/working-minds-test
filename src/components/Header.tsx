import styled from 'styled-components';
import { Logo } from './Logo';

const HeaderContainer = styled.header`
  width: 100%;
  background-color: #51bbfe;
  height: 50px;
  display: flex;
  align-items: center;
  padding: 1rem;
  justify-content: space-between;
  color: #f4f4f4;
`;

export function Header(): JSX.Element {
  return (
    <HeaderContainer>
      <Logo />
    </HeaderContainer>
  );
}
