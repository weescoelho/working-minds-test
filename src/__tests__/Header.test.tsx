import { create } from 'react-test-renderer';
import { Header } from '../components/Header';

test('render Header component', () => {
  const c = create(<Header />);
  expect(c.toJSON()).toMatchSnapshot();
});
