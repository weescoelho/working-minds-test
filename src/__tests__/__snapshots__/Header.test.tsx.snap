// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header component 1`] = `
<header
  className="sc-gtsrHT cAQDTl"
>
  <div
    className="sc-bdnxRM WOKuE"
  >
    CRUD
  </div>
</header>
`;
